pkgs/stdenv/generic/default.nix

```diff
@@ -147,7 +161,15 @@ let
           nativeBuildInputs = nativeBuildInputs ++ (if crossConfig == null then buildInputs else []);
           propagatedNativeBuildInputs = propagatedNativeBuildInputs ++
             (if crossConfig == null then propagatedBuildInputs else []);
-        })) (
+
+          __impureHostDeps = lib.unique (lib.sort (x: y: x < y) (computedImpureHostDeps ++ computedPropagatedImpureHostDeps ++ __propagatedImpureHostDeps ++ __impureHostDeps ++ __extraImpureHostDeps ++ [
```

Comment:
@edolstra is there a better way to canonicalize lists of things? It seems like a generally useful construct, for cases in which we don't actually care about the order of things, like this (and probably several other places). I don't mind dropping that stuff right now, but perhaps a primop or two (with and without `uniq`) to do this kind of thing would be good?